lib/Objects/basic_device.dart

```diff
@@ -0,0 +1,40 @@
+class BasicDevice {
+  String kind;
+  String etag;
+  String deviceId;
+  String serialNumber;
+  String status;
+  String lastSync;
+  String supportEndDate;
+  String annotatedUser;
+  String annotatedLocation;
+  String annotatedAssetId;
+  String notes;
+
+  BasicDevice(
+      this.kind,
```

Comment:
You should make this have all optional args too, with the {} syntax.  You can specify specific items as required if needed to by putting in the @required annotation on the items, you can also set defaults using this syntax too.